homeassistant/components/motioneye/switch.py

```diff
@@ -67,38 +84,37 @@ class MotionEyeSwitch(MotionEyeEntity, SwitchEntity):
 
     def __init__(
         self,
+        entity_description: EntityDescription,
         config_entry_id: str,
         camera: dict[str, Any],
-        switch_key: str,
-        switch_key_friendly_name: str,
         client: MotionEyeClient,
         coordinator: DataUpdateCoordinator,
         options: MappingProxyType[str, str],
-        enabled_by_default: bool,
     ) -> None:
         """Initialize the switch."""
-        self._switch_key = switch_key
-        self._switch_key_friendly_name = switch_key_friendly_name
+        self._entity_description = entity_description
```

Comment:
You can set it on the entity like so:

```suggestion
        self.entity_description = entity_description
```

The entity will automatically pick up `entity_registry_enabled_default` from the entity description in that case (so you don't need to pass it down anymore).